catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/SearchFormsPredicatesIT.java

```diff
@@ -0,0 +1,367 @@
+          null, // No interaction
+          MOCK_CONFIG);
+
+  private static final SearchFormsApplication APPLICATION =
+      new SearchFormsApplication(
+          MOCK_FRAMEWORK, FILTER_BUILDER, TRANSFORMER, UTIL, () -> MOCK_SUBJECT);
+
+  // Will be initialized by setUpClass() when the port is known
+  private static String localhostFormsUrl = null;
+
+  /*
+   * ---------------------------------------------------------------------------------------------
+   * Test Exec
+   * ---------------------------------------------------------------------------------------------
+   */
+
+  @BeforeClass
+  public static void setUpClass() {
+    Spark.port(getAvailablePort());
+    APPLICATION.init();
+    Spark.awaitInitialization();
+    localhostFormsUrl = format("http://localhost:%d/forms/query", Spark.port());
+  }
+
+  @AfterClass
+  public static void tearDownClass() {
+    stop();
+  }
+
+  @Before
+  public void setUp() {
+    when(MOCK_SUBJECT.isGuest()).thenReturn(false);
+    when(MOCK_CONFIG.getMaximumUploadSize()).thenReturn(1024);
+  }
+
+  @After
+  public void tearDown() {
+    reset(MOCK_FRAMEWORK, MOCK_SUBJECT, MOCK_CONFIG);
+  }
+
+  @Test
+  public void testTextEqualJsonToXml() throws IngestException, SourceUnavailableException {
+    testJsonToXml(
+        getJsonFilter("=", "language", "english"),
+        getXmlFilter("PropertyIsEqualTo", "language", "english"));
+  }
+
+  @Test
+  public void testTextEqualXmlToJson()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+    testXmlToJson(
+        getXmlFilter("PropertyIsEqualTo", "language", "english"),
+        getJsonResponse("=", "language", "english"));
+  }
+
+  @Test
+  public void testDateBetweenJsonToXml() throws IngestException, SourceUnavailableException {
+    testJsonToXml(
+        getJsonFilter("DURING", "created", CANNED_ISO_DATE + "/" + CANNED_ISO_DATE),
+        getXmlFilter("During", "created", CANNED_EPOCH_DATE + "/" + CANNED_EPOCH_DATE));
+  }
+
+  @Test
+  public void testDateBetweenXmlToJson()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+    testXmlToJson(
+        getXmlFilter("During", "created", CANNED_EPOCH_DATE + "/" + CANNED_EPOCH_DATE),
+        getJsonResponse(
+            "DURING",
+            "created",
+            CANNED_ISO_DATE + "/" + CANNED_ISO_DATE,
+            CANNED_ISO_DATE,
+            CANNED_ISO_DATE));
+  }
+
+  @Test
+  public void testDateEmptyJsonToXml() throws IngestException, SourceUnavailableException {
+    testJsonToXml(
+        // counteract whitespace removal
+        getJsonFilter("IS NULL", "created", null).replace("ISNULL", "IS NULL"),
+        getNilXmlFilter("created"));
+  }
+
+  @Test
+  public void testDateEmptyXmlToJson()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+    testXmlToJson(
+        getNilXmlFilter("created"),
+        // counteract whitespace removal
```

Comment:
❓  why does the whitespace get removed?

Reply:
The test data is prettified so it can be read / reviewed with ease. I have some functions, one for XML and one for JSON, that removes the pretty printing so I can do a simple string equality comparison for the test. Also, so the XML accurately matches how it looks on the metacard attribute in Solr at the time of the test. 

Problem is that one of the operators is defined as `"IS EMPTY"` and not `IS_EMPTY` or some other symbol-delineated form. So for this test, and only this test, I need to fix it. It's only there for the sake of the test and doesn't impact prod code.